### ISSUE TITLE
parallelize metadata collection loop

### ISSUE THREAD

**Issue by carloshorn** [pytroll/pygac-fdr#60]
When collecting the metadata, the following loop consumes most of the processing time and would benefit from a parallelisation.
https://github.com/pytroll/pygac-fdr/blob/4e52f7d133ee9c0541de01307651403de4a28ae6/pygac_fdr/metadata.py#L157-L181

In combination with #31 this could be implemented using
```python
import multiprocessing

[...]
with multiprocessing.Pool(n_worker) as pool:
    for metadata in pool.imap_unordered(extract_metadata, filenames):
        session.add(metadata)
        [...]
```

**Comment by sfinkens** [comment]
Good idea! `pygac-fdr-run` would also benefit from that probably (processing multiple files in parallel).
